File: problems/18_bit_manipulation/0136_single_number/solution.dart
// 0136. Single Number
int singleNumber(List<int> nums) {
  var x = 0;
  for (final v in nums) x ^= v;
  return x;
}
void main() {
  if (singleNumber([2, 2, 1]) != 1) throw 'a';
  if (singleNumber([4, 1, 2, 1, 2]) != 4) throw 'b';
  if (singleNumber([1]) != 1) throw 'c';
  if (singleNumber([7, 3, 5, 4, 5, 3, 4]) != 7) throw 'd';
  print('All tests passed!');
}

File: problems/18_bit_manipulation/0136_single_number/solution.py
"""
Problem: 0136. Single Number
"""
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        x = 0
        for v in nums:
            x ^= v
        return x


if __name__ == "__main__":
    sol = Solution()
    assert sol.singleNumber([2, 2, 1]) == 1
    assert sol.singleNumber([4, 1, 2, 1, 2]) == 4
    assert sol.singleNumber([1]) == 1
    assert sol.singleNumber([7, 3, 5, 4, 5, 3, 4]) == 7
    print("All tests passed!")
